refactor kalshiAPIUtil

i16n opened the pull request:
refactored logic in kalshiAPIUtil, just extracting helper methods

i16n committed:
refactor kalshiAPIUtil

ay3xqa commented on server/kalshiAPIUtil.py:
```diff
@@ -42,105 +42,161 @@ def get_kalshi_max_year_json(currency, SMA):
-        return {}
+        return None
+        
     if response.status_code == 200:
```
lets do >299 and treat that as error else: we do our happy case

ay3xqa reviewed (commented):
Lets run this, and also lets run the client (react app ui) as well because the json created in this module is the json payload sent to the react ui

i16n committed:
fix small logic and logging issue

i16n committed:
fix parameter issue

i16n commented:
also added some logging changes just for better visual understanding, and found one issue with the refactor regarding an unused parameter, so I fixed that too. Fixed the issue detailed above. Ran the front end and it appears to be working.

ay3xqa reviewed (approved):
LGTM
